### PR TITLE
[pango] Fix the dependency type name of freetype

### LINKED ISSUE #27801: [pango[core]:x64-windows-static] build failure
Package: pango[core]:x64-windows-static -> 1.50.11

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.34.31933.0
-    vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
    vcpkg-scripts version: 6f7ffeb18 2022-11-11 (2 days ago)

**To Reproduce**

`vcpkg install pango:x64-windows-static`

**Failure logs**

```
-- Downloading https://gitlab.gnome.org//GNOME/pango/-/archive/eabdbd54ee38fa658be574c4fa4574a05c755f16/pango-eabdbd54ee38fa658be574c4fa4574a05c755f16.tar.gz -> GNOME-pango-eabdbd54ee38fa658be574c4fa4574a05c755f16.tar.gz...
-- Extracting source T:/Dev/test/vcpkg/downloads/GNOME-pango-eabdbd54ee38fa658be574c4fa4574a05c755f16.tar.gz
-- Using source at T:/Dev/test/vcpkg/buildtrees/pango/src/a05c755f16-d464bd60e8.clean
CMake Warning at scripts/cmake/vcpkg_configure_meson.cmake:305 (message):
  Options ADDITIONAL_(NATIVE|CROSS)_BINARIES have been deprecated.  Only use
  ADDITIONAL_BINARIES!
Call Stack (most recent call first):
  ports/pango/portfile.cmake:10 (vcpkg_configure_meson)
  scripts/ports.cmake:147 (include)


-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows-static
-- Downloading https://www.python.org/ftp/python/3.10.5/python-3.10.5-embed-amd64.zip -> python-3.10.5-embed-amd64.zip...
-- Configuring x64-windows-static-dbg
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://www2.futureware.at/~nickoe/msys2-mirror/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.yandex.ru/mirrors/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.tuna.tsinghua.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.ustc.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.bit.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirror.selfnet.de/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst;https://mirrors.sjtug.sjtu.edu.cn/msys2/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst -> msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://www2.futureware.at/~nickoe/msys2-mirror/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.yandex.ru/mirrors/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.tuna.tsinghua.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.ustc.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.bit.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirror.selfnet.de/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://mirrors.sjtug.sjtu.edu.cn/msys2/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst -> msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst...
-- Using msys root at T:/Dev/test/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x64-windows-static-dbg done
-- Configuring x64-windows-static-rel
-- Configuring x64-windows-static-rel done
-- Package x64-windows-static-dbg
-- Package x64-windows-static-rel
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pango.pc
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pangocairo.pc
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pangofc.pc
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pangoft2.pc
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pangoot.pc
-- Fixing pkgconfig file: T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig/pangowin32.pc
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:31 (message):
  
  T:/Dev/test/vcpkg/downloads/tools/msys2/9a1ec3f33446b195/mingw32/bin/pkg-config.exe
  --exists pangocairo failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig;T:/Dev/test/vcpkg/packages/pango_x64-windows-static/share/pkgconfig;T:/Dev/test/vcpkg/installed/x64-windows-static/lib/pkgconfig;T:/Dev/test/vcpkg/installed/x64-windows-static/share/pkgconfig"
      output: Empty package name in Requires or Conflicts in file 'T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig\pangoft2.pc'
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:154 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/pango/portfile.cmake:28 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:147 (include)



```

**Additional context**

Entire logs at T:\Dev\test\vcpkg\buildtrees\pango are attached.
[logs.zip](https://github.com/microsoft/vcpkg/files/10001494/logs.zip)

### PR DESCRIPTION
**Pull request by LilyWangLL**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #27801, change `freetype_dep.type_name()` to cmake, add `freetype2` to `Requires` in **pangoft2.pc**, and fix the build error:
```
ENV{PKG_CONFIG_PATH}: "T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig;T:/Dev/test/vcpkg/packages/pango_x64-windows-static/share/pkgconfig;T:/Dev/test/vcpkg/installed/x64-windows-static/lib/pkgconfig;T:/Dev/test/vcpkg/installed/x64-windows-static/share/pkgconfig"
output: Empty package name in Requires or Conflicts in file 'T:/Dev/test/vcpkg/packages/pango_x64-windows-static/lib/pkgconfig\pangoft2.pc'
```
The original `Requires` content is as below, about `freetype` is empty.
x64-windows:
```
Requires: pango, , fontconfig >=  2.13.0
Requires.private: glib-2.0 >=  2.62, gobject-2.0 >=  2.62, gio-2.0 >=  2.62, fribidi >=  1.0.6, harfbuzz >=  2.6.0, cairo >=  1.12.10
```
x64-windows-static:
```
Requires: pango, glib-2.0 >=  2.62, gobject-2.0 >=  2.62, gio-2.0 >=  2.62, fribidi >=  1.0.6, harfbuzz >=  2.6.0, fontconfig >=  2.13.0, cairo >=  1.12.10, 
```

### CODE REVIEW

**Comment by LilyWangLL** [comment]
cc @Neumann-A Could you help to review this PR? And I don't know this bug is a upstream bug or not. Need I submit an issue on [upstream](https://gitlab.gnome.org/GNOME/pango/-/issues) of `pango`? Thanks in advance.

File: ports/pango/portfile.cmake [hunk 1]
```diff
@@ -18,10 +20,10 @@ vcpkg_configure_meson(
...
         -Dfreetype=enabled # Build with freetype support
         -Dgtk_doc=false #Build API reference for Pango using GTK-Doc
-    ADDITIONAL_NATIVE_BINARIES glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
-    ADDITIONAL_CROSS_BINARIES  glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
+    ADDITIONAL_BINARIES glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
+                        glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
+                        glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
+                        glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
```
Comment (Neumann-A) [inline ports/pango/portfile.cmake:26]:
```suggestion
```

File: ports/pango/fix-pangoft2PC.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/meson.build b/meson.build
+index 4e796ec..9693dc7 100644
+--- a/meson.build
++++ b/meson.build
+@@ -345,7 +345,7 @@ freetype_dep = dependency(freetype_package_name,
+                           required: freetype_option,
+                           fallback: ['freetype2', 'freetype_dep'])
+ 
+-if freetype_dep.found() and freetype_dep.type_name() == 'pkgconfig'
++if freetype_dep.found() and freetype_dep.type_name() == 'cmake'
```
Comment (Neumann-A) [inline ports/pango/fix-pangoft2PC.patch:10]: why is this needed ?

Comment (LilyWangLL) [reply ports/pango/fix-pangoft2PC.patch:10]: You can see the description of this PR.

Comment (Neumann-A) [reply ports/pango/fix-pangoft2PC.patch:10]: https://github.com/GNOME/pango/blob/d616f88dbb6dd5360eebf8ab42884fb90583218b/meson.build#L243-L246 should probably be `[]` and not `''` and you probably can remove the if in. https://github.com/GNOME/pango/blob/d616f88dbb6dd5360eebf8ab42884fb90583218b/meson.build#L315

Comment (Neumann-A) [inline ports/pango/fix-pangoft2PC.patch:10]:
```suggestion
+if freetype_dep.found()
```

thinking about it just removing the second part seems more reasonable.

Comment (LilyWangLL) [reply ports/pango/fix-pangoft2PC.patch:10]: I don't know why the type name of `freetype2` isn't `pkgconfig`. It has PC file.

Comment (Neumann-A) [reply ports/pango/fix-pangoft2PC.patch:10]: because on windows it looks for `freetype` and not `freetype2` which is the name of the pkg-config file

Comment (Neumann-A) [reply ports/pango/fix-pangoft2PC.patch:10]: see https://github.com/GNOME/pango/blob/d616f88dbb6dd5360eebf8ab42884fb90583218b/meson.build#L315

**Comment by LilyWangLL** [comment]
Closed as duplicate of https://github.com/microsoft/vcpkg/pull/28766
